front/src/common/Map/Search/MapSearchSignal.tsx

```diff
@@ -0,0 +1,263 @@
+  const [aspect, setAspect] = useState<string[]>([]);
+  const [searchResults, setSearchResults] = useState<ISignalSearchResult[]>([]);
+  // Sort by, and order true = ASC, false = DESC
+  const [sortFilter, setSortFilter] = useState<SortType>({
+    name: 'label',
+    asc: true,
+  });
+  const dispatch = useDispatch();
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const { t } = useTranslation(['translation', 'map-search']);
+
+  const getPayload = (lineSearch: string, signalName: string) => {
+    const payloadQuery: (string | number | string[])[] = !Number.isNaN(Number(lineSearch))
+      ? ['=', ['line_code'], Number(lineSearch)]
+      : ['search', ['line_name'], lineSearch];
+
+    const aspectQuery = aspect.join(' ');
+    return {
+      object: 'signal',
+      query: [
+        'and',
+        ['=', ['infra_id'], infraID],
+        payloadQuery,
+        !aspectQuery || ['contains', ['list', aspectQuery], ['aspects']],
+        ['search', ['label'], signalName],
+      ],
+    };
+  };
+
+  const updateSearch = async () => {
+    const payload: searchPayloadType = getPayload(searchLineState, searchState);
+    await postSearch({
+      body: payload,
+    })
+      .unwrap()
+      .then((results) => {
+        setSearchResults([...results] as ISignalSearchResult[]);
+      })
+      .catch((e) => {
+        console.error(e);
+        setSearchResults([]);
+        dispatch(
+          setFailure({
+            name: t('map-search:errorMessages.unableToSearchSignal'),
+            message: `${e.message}`,
+          })
+        );
+      });
+  };
+
+  const debouncedSearchTerm = useDebounce(searchState, 300);
+  const debouncedSearchLine = useDebounce(searchLineState, 300);
+
+  useEffect(() => {
+    if (searchLineState) {
+      updateSearch();
+    } else {
+      setSearchResults([]);
+    }
+  }, [debouncedSearchTerm, debouncedSearchLine, aspect]);
+
+  const onResultClick = (result: ISignalSearchResult) => {
+    const coordinates = getCoordinates(result, map);
+
+    const center = turfCenter(coordinates as AllGeoJSON);
+
+    const newViewport = {
+      ...map.viewport,
+      longitude: center.geometry.coordinates[0],
+      latitude: center.geometry.coordinates[1],
+      zoom: 12,
+    };
+    updateExtViewport(newViewport);
+    dispatch(
+      updateMapSearchMarker({
+        title: result.label,
+        lonlat: [center.geometry.coordinates[0], center.geometry.coordinates[1]],
+      })
+    );
+  };
+
+  const formatSearchResults = () => {
+    const searchResultsContent = searchResults.sort((a, b) => {
+      if (!a[sortFilter.name]) {
+        return sortFilter.asc ? 1 : -1;
+      } // To avoid null values
+      if (!b[sortFilter.name]) {
+        return sortFilter.asc ? -1 : 1;
+      }
+      return sortFilter.asc
+        ? String(b[sortFilter.name]).localeCompare(String(a[sortFilter.name]))
+        : String(a[sortFilter.name]).localeCompare(String(b[sortFilter.name]));
+    });
```

Comment:
```suggestion
```

You can replace this part of the function by a useEffect like this (use sortBy function of lodash): 
```
useEffect(() => {
    const sortedResults = sortBy(searchResults, sortFilter.name);
    if (sortFilter.asc) {
      setSearchResults(sortedResults.reverse());
    } else {
      setSearchResults(sortedResults);
    }
  }, [sortFilter]);
```

Reply:
Cool !! Thank you for a good idea

Reply:
Done